api/cluster/pdb.go

```diff
@@ -0,0 +1,113 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/caraml-dev/merlin/config"
+	"github.com/caraml-dev/merlin/models"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	metav1cfg "k8s.io/client-go/applyconfigurations/meta/v1"
+	policyv1cfg "k8s.io/client-go/applyconfigurations/policy/v1"
+)
+
+type PodDisruptionBudget struct {
+	Name                     string
+	Namespace                string
+	Labels                   map[string]string
+	MaxUnavailablePercentage *int
+	MinAvailablePercentage   *int
+	Selector                 *metav1.LabelSelector
+}
+
+func NewPodDisruptionBudget(modelService *models.Service, componentType string, pdbConfig config.PodDisruptionBudgetConfig) *PodDisruptionBudget {
+	return &PodDisruptionBudget{
+		Name:                     fmt.Sprintf("%s-%s-%s", modelService.Name, componentType, models.PDBComponentType),
+		Namespace:                modelService.Namespace,
+		Labels:                   modelService.Metadata.ToLabel(),
+		MaxUnavailablePercentage: pdbConfig.MaxUnavailablePercentage,
+		MinAvailablePercentage:   pdbConfig.MinAvailablePercentage,
+	}
+}
+
+func (cfg PodDisruptionBudget) BuildPDBSpec() (*policyv1cfg.PodDisruptionBudgetSpecApplyConfiguration, error) {
+	if cfg.MaxUnavailablePercentage == nil && cfg.MinAvailablePercentage == nil {
+		return nil, fmt.Errorf("one of maxUnavailable and minAvailable must be specified")
+	}
+
+	pdbSpec := &policyv1cfg.PodDisruptionBudgetSpecApplyConfiguration{
+		Selector: &metav1cfg.LabelSelectorApplyConfiguration{
+			MatchLabels: cfg.Labels,
+		},
+	}
+
+	if cfg.MaxUnavailablePercentage != nil {
```

Comment:
Same nit comment about the logic as in Turing: https://github.com/caraml-dev/turing/pull/346/files#r1239515217